sparkly/testing.py

```diff
@@ -135,13 +172,294 @@ def assertDataFrameEqual(self, actual_df, expected_data, fields=None, ordered=Fa
         actual_rows = actual_df.collect() if hasattr(actual_df, 'collect') else actual_df
         actual_data = [row.asDict(recursive=True) for row in actual_rows]
 
-        if ordered:
-            self.assertEqual(actual_data, expected_data)
-        else:
-            try:
-                self.assertCountEqual(actual_data, expected_data)
-            except AttributeError:
-                self.assertItemsEqual(actual_data, expected_data)
+        return self.assertEqual(
+            actual_data,
+            expected_data,
+            ignore_order=not ordered,
+            ignore_order_depth=1,
+            rtol=0,
+            equal_nan=False,
+        )
+
+    def assertEqual(self,
```

Comment:
How about `assertDataFrameEqual`? There is no need to make it god-like function that manages everything. We have `assertAlmostEqual` to compare `float` and `assertCountEqual` to compare `list` without preserving the order.